packages/mobile/src/app/saga.ts

```diff
@@ -50,15 +51,18 @@ export function* appInit() {
     yield put(setLanguage(language))
   }
 
-  const deepLink = yield call(Linking.getInitialURL)
+  const deepLink: string | null = yield call(Linking.getInitialURL)
   const inSync = yield call(clockInSync)
   if (!inSync) {
     navigate(Screens.SetClock)
     return
   }
 
   if (deepLink) {
-    handleDeepLink(deepLink)
+    // TODO: this should dispatch (put) but since this appInit
+    // is called before the listener is set, we do it this way.
+    // This is fragile, change me :D
```

Comment:
Why should this put instead of call?

Reply:
Ah yes my thought was that the `openDeepLink` action should be dispatched to follow the same path as handling deep links when the app is already open.
This is to avoid surprises and have a consistent behavior (maybe a reducer does something with the action).